py/desispec/coaddition.py

```diff
@@ -198,12 +219,20 @@ def coadd_fibermap(fibermap, onetile=False):
             log.error(msg)
             raise ValueError(msg)
 
+    #- make a copy of input fibermap that we can modify with new columns
+    fibermap = Table(fibermap, copy=True)
```

Comment:
Do units get dropped with any of these techniques?

Reply:
@akremin we no longer need to check the type of fibermap at the end when doing column sub-selection because it was already copied/cast to a Table at the very beginning.

`fibermap` not being called `exp_fibermap` was a historic leftover and I originally hadn't changed it here to focus the PR changes on algorithmic differences rather than renaming cleanup, but I agree that it is confusing that the variable called "fibermap" internal to the function is destined for the HDU called "EXP_FIBERMAP", while a different variable "tfmap" becomes the "FIBERMAP" HDU.  So as a final commit to this PR I did that renaming change at the beginning rather than the end.  Good idea.

@weaverba137 I confirmed on main and this branch that neither method drops units (those are getting dropped at some earlier step), but I added a new unit test to confirm that so that we don't accidentally drop units in some future coadd_fibermap update.

Reply:
I understood we no longer needed the lines at the end of the function, my question was whether your simplified copy scheme is as robust as the old code, given that it used to be wrapped in an if/else and isn't in your implementation. My assumption is that you got rid of the if/else because either `astropy.Table` or `desispec` have been made robust enough that the statement is unnecessary, but I wanted to check that explicitly.

Reply:
Got it.  Previously I was trying to be overly fancy with avoiding a double copy when casting to a Table and sub-selecting columns while ensuring that the columns themselves were copies and not just views of the underlying arrays.  The new code makes a full single copy at the very start when casting to a Table, so it is robust to whether the input was originally a Table or a numpy structured array.  Good to check, thanks.

Reply:
Good idea checking units anyway!